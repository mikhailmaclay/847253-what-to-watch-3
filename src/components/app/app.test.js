// Libraries
import React from 'react';
import renderer from 'react-test-renderer';
// Components
import AppContainer from './app.container';
import {createStore} from 'redux';
import {Provider} from 'react-redux';

const movies = [
  {
    id: 1,
    name: `The Grand Budapest Hotel`,
    genre: `Drama`,
    releaseDate: `2014`,
    reviews: [1],
    images: [`/img/the-grand-budapest-hotel.jpg`],
    poster: `/img/the-grand-budapest-hotel-poster.jpg`,
    background: `/img/bg-the-grand-budapest-hotel.jpg`
  }
];

const reviews = [
  {
    id: 1,
    user: 1,
    rating: 5,
    date: `November 18, 2020 03:24:00`,
    text: `DO! IT!`
  }
];

const users = [
  {id: 1, fullName: `Shia LaBeouf`}
];

const initialState = {
  movies,
  reviews,
  users,
  specialMovie: 1
};

const store = createStore((state) => state, initialState);

describe(`<App/>`, () => {
  it(`should be rendered correctly`, () => {
    const result = renderer.create(
        <Provider store={store}>
          <AppContainer/>
        </Provider>
    ).toJSON();

    expect(result).toMatchSnapshot();
  });
});
