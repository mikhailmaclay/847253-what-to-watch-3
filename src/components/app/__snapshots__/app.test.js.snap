// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> should be rendered correctly 1`] = `
Array [
  <div
    className="movie-card__bg"
  />,
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>,
  <header>
    <div
      className="logo"
    >
      <a
        className="logo__link "
        href={null}
        onClick={null}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="user-block"
    >
      <a
        className="user-block__link"
        href="/sign-in"
        onClick={[Function]}
      >
        Sign in
      </a>
    </div>
  </header>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <button
            className="catalog__genres-link"
            disabled={true}
            onClick={null}
            style={
              Object {
                "background": "transparent",
                "border": "none",
                "cursor": "",
                "paddingLeft": "0",
                "paddingRight": "0",
                "userSelect": "none",
              }
            }
            type="button"
          >
            All genres
          </button>
        </li>
      </ul>
      Sorry, there is no movies.
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href={null}
          onClick={null}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
